From 58a0d4031de50273a316bc6c7ed9728efd0218d3 Mon Sep 17 00:00:00 2001
From: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
Date: Wed, 16 Sep 2020 14:24:50 -0600
Subject: [PATCH 1594/1851] memory: pl353: Fixed incompatible assignment

commit 169d9d4cc798d28c62e80b2da2adaeb5000a6451 from
https://github.com/Xilinx/linux-xlnx.git

Fixed incompatible assignment.

Addresses-coverity: "incompatible_assignment_operands"
Signed-off-by: Amit Kumar Mahapatra <amit.kumar-mahapatra@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/memory/pl353-smc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/memory/pl353-smc.c b/drivers/memory/pl353-smc.c
index 73bd3023202f..f90e50e1078b 100644
--- a/drivers/memory/pl353-smc.c
+++ b/drivers/memory/pl353-smc.c
@@ -411,7 +411,8 @@ static int pl353_smc_probe(struct amba_device *adev, const struct amba_id *id)
 		goto out_clk_disable;
 	}
 
-	init = match->data;
+	init = (void (*)(struct amba_device *,
+			 struct device_node *))match->data;
 	if (init)
 		init(adev, child);
 	of_platform_device_create(child, NULL, &adev->dev);
-- 
2.31.1

